### ISSUE TITLE
BUILD_FAILED WINDOWS 11

### ISSUE THREAD

**Issue by VadimS077** [microsoft/vcpkg#29798]
**Describe the bug**

**Environment**
- OS: WINDOWS 11
- Compiler: revision

**To Reproduce**
1. vcpkg install doctes
2. See error
![image](https://user-images.githubusercontent.com/110056050/220752269-2d45094a-2719-4ba6-83cf-101494e06058.png)

So,i have last version of all,what i have. It's vcpkg,git cmd. I did all steps,which are on vcpkg.io. I started to install doctest,but everytime  i get this error. So,i dont't have a proxy or something like that,but even if i have,i have automatic setting on Windows 11. Also, there isn't problem with another programs (proxy problem). Help please

**Comment by talregev** [comment]
Did you install visual studio?
Also please copy and paste the errors, don't print screen it. It hard to understand.
Also you should upload to here, or copy paste the logs that it point out in the error msg.
If you do that people will help you here.

**Comment by FrankXie05** [comment]
@VadimS077 Please ensure that your network is unblocked and if you use a proxy, please ensure that vcpkg passes the proxy.
BTW, could you access https://github.com/doctest/doctest/archive/v2.4.9.tar.gz (valid) manually?

**Comment by VadimS077** [comment]
> 

error: building doctest:x86-windows failed with: BUILD_FAILED
error: Please ensure you're using the latest port files with `git pull` and `vcpkg update`.
Then check for known issues at:
    https://github.com/microsoft/vcpkg/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+doctest
You can submit a new issue at:
    https://github.com/microsoft/vcpkg/issues/new?template=report-package-build-failure.md&title=[doctest]+Build+error
Include '[doctest] Build error' in your bug report title, the following version information in your bug description, and attach any relevant failure logs from above.
    vcpkg-tool version: 2023-01-24-8a88d63f241d391772fbde69af9cab96c3c64c75
    vcpkg-scripts version: d098bf1e8 2023-02-22 (14 minutes ago)
There are errors. Yeah,i have visual studio and i have done "integrate install"

**Comment by VadimS077** [comment]
> @VadimS077 Please ensure that your network is unblocked and if you use a proxy, please ensure that vcpkg passes the proxy. BTW, could you access https://github.com/doctest/doctest/archive/v2.4.9.tar.gz (valid) manually?

Yes, i can it download manually. And i did that and copy it on vcpkg/downloads and try again "vcpkg install doctest" but again i have same problem

**Comment by talregev** [comment]
Can you also try?
`ping  github.com`

**Comment by VadimS077** [comment]
> Can you also try? `ping github.com`

Pinging github.com [140.82.121.3] with 32 bytes of data:
Reply from 140.82.121.3: bytes=32 time=46ms TTL=53
Reply from 140.82.121.3: bytes=32 time=43ms TTL=53
Reply from 140.82.121.3: bytes=32 time=44ms TTL=53
Reply from 140.82.121.3: bytes=32 time=43ms TTL=53

Ping statistics for 140.82.121.3:
    Packets: Sent = 4, Received = 4, Lost = 0 (0% loss),
Approximate round trip times in milli-seconds:
    Minimum = 43ms, Maximum = 46ms, Average = 44ms

**Comment by talregev** [comment]
> > Can you also try? `ping github.com`
> 
> Pinging github.com [140.82.121.3] with 32 bytes of data: Reply from 140.82.121.3: bytes=32 time=46ms TTL=53 Reply from 140.82.121.3: bytes=32 time=43ms TTL=53 Reply from 140.82.121.3: bytes=32 time=44ms TTL=53 Reply from 140.82.121.3: bytes=32 time=43ms TTL=53
> 
> Ping statistics for 140.82.121.3: Packets: Sent = 4, Received = 4, Lost = 0 (0% loss), Approximate round trip times in milli-seconds: Minimum = 43ms, Maximum = 46ms, Average = 44ms

If ping is working, you can try to do it with curl:
`curl -L https://github.com/doctest/doctest/archive/v2.4.9.tar.gz --output v2.4.9.tar.gz`

curl is a build in tool in windows11. 

If curl is working and vcpkg not, then there is a problem connected in vcpkg.

**Comment by VadimS077** [comment]
C:\Users\vcpkg>curl -L https://github.com/doctest/doctest/archive/v2.4.9.tar.gz --output v2.4.9.tar.gz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:--  0:00:01 --:--:--     0
curl: (60) schannel: SNI or certificate check failed: SEC_E_WRONG_PRINCIPAL (0x80090322)
More details here: https://curl.se/docs/sslcerts.html

curl failed to verify the legitimacy of the server and therefore could not
establish a secure connection to it. To learn more about this situation and
how to fix it, please visit the web page mentioned above.

**Comment by talregev** [comment]
> C:\Users\vcpkg>curl -L https://github.com/doctest/doctest/archive/v2.4.9.tar.gz --output v2.4.9.tar.gz % Total % Received % Xferd Average Speed Time Time Time Current Dload Upload Total Spent Left Speed 0 0 0 0 0 0 0 0 --:--:-- --:--:-- --:--:-- 0 0 0 0 0 0 0 0 0 --:--:-- 0:00:01 --:--:-- 0 curl: (60) schannel: SNI or certificate check failed: SEC_E_WRONG_PRINCIPAL (0x80090322) More details here: https://curl.se/docs/sslcerts.html
> 
> curl failed to verify the legitimacy of the server and therefore could not establish a secure connection to it. To learn more about this situation and how to fix it, please visit the web page mentioned above.

We found the problem. Not sure how to solve it. 
I just google the problem. there is an easy solution on Linux. 
I am not sure how to update certificates on windows.

**Comment by talregev** [comment]
Maybe is your clock? Check your time is setup correctly.

**Comment by VadimS077** [comment]
> Maybe is your clock? Check your time is setup correctly.

clock is correct,i even update it again with windows settings

**Comment by VadimS077** [comment]
It seems like this is problem with windows

**Comment by VadimS077** [comment]
I noticed the fact that when using a vpn I have access to the site https://github.com/doctest/doctest/archive/v2.4.9.tar.gz , but without vpn , the following error crashes:Your connection is not private
Attackers may try to steal your data from codeload.github.com (for example, passwords, messages, or credit card details).

NET::ERR_CERT_COMMON_NAME_INVALID

**Comment by talregev** [comment]
try to download the file with curl when your vpn is open and close.

**Comment by VadimS077** [comment]
> try to download the file with curl when your vpn is open and close.

it works with vpn integrated in browser. With my free vpn on pc it doesn't work

**Comment by VadimS077** [comment]
> > try to download the file with curl when your vpn is open and close.
> 
> it works with vpn integrated in browser. With my free vpn on pc it doesn't work

but vcpkg don't see integrated vpn

**Comment by talregev** [comment]
> it works with vpn integrated in browser

It work with curl?
`curl -L https://github.com/doctest/doctest/archive/v2.4.9.tar.gz --output v2.4.9.tar.gz`?

**Comment by VadimS077** [comment]
> > it works with vpn integrated in browser
> 
> It work with curl? `curl -L https://github.com/doctest/doctest/archive/v2.4.9.tar.gz --output v2.4.9.tar.gz`?

no(

**Comment by VadimS077** [comment]
Website open only with integrated vpn. But curl and vcpkg don't work

**Comment by talregev** [comment]
I am not sure it good direction, but you can try to configure connection on windows that will work with your vpn.

https://support.microsoft.com/en-us/windows/connect-to-a-vpn-in-windows-3d29aeb1-f497-f6b7-7633-115722c1009c#:~:text=Select%20the%20Start%20button%2C%20then%20select%20Settings%20%3E%20Network%20%26%20Internet,VPN%20%3E%20Add%20a%20VPN%20connection.

When you download with curl, you will download with vcpkg. Do it with curl, show you more errors.

**Comment by VadimS077** [comment]
Unfortunately,there isn't working vpn for https://github.com/doctest/doctest/archive/v2.4.9.tar.gz
I tried many vpn profiles and browser said,that it really works
![image](https://user-images.githubusercontent.com/110056050/220954268-fec298ef-2032-4ac7-b817-5aa2110b7d99.png)
But i have same problems with curl

**Comment by VadimS077** [comment]
And it is strange,because with integrated vpn(1 click it calls) i can open https://github.com/doctest/doctest/archive/v2.4.9.tar.gz

**Comment by talregev** [comment]
I think it also certificate problem. because when you show curl output, it say that it complain about the certificate. 
I am not expert in it, so I will wait for other that will try to help you.
Also, if you have new input,  you can put it here. If I know something I will comment again.

**Comment by talregev** [comment]
@FrankXie05 Do you have some direction?

**Comment by VadimS077** [comment]
When i use C:\Users\user>curl -k https://github.com/doctest/doctest/archive/v2.4.9.tar.gz --output v2.4.9.tar.gz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:--  0:00:03 --:--:--     0
-k disable certificate, but i don't get anything

**Comment by VadimS077** [comment]
Also i tried the same on my PC and it works. So the problem is local with my laptop

**Comment by talregev** [comment]
> When i use C:\Users\user>curl -k https://github.com/doctest/doctest/archive/v2.4.9.tar.gz --output v2.4.9.tar.gz % Total % Received % Xferd Average Speed Time Time Time Current Dload Upload Total Spent Left Speed 0 0 0 0 0 0 0 0 --:--:-- 0:00:03 --:--:-- 0 -k disable certificate, but i don't get anything

Look the files in the folder: 
`dir` 
Do you see the file download? 
Also I am not sure if vcpkg there is an option to download with ignoring the certificate. Also it not a safe option.

**Comment by talregev** [comment]
> Also i tried the same on my PC and it works. So the problem is local with my laptop

Is it a laptop from work? Usually the have vpn and configure that it hard to download stuff inside.

**Comment by VadimS077** [comment]
> > When i use C:\Users\user>curl -k https://github.com/doctest/doctest/archive/v2.4.9.tar.gz --output v2.4.9.tar.gz % Total % Received % Xferd Average Speed Time Time Time Current Dload Upload Total Spent Left Speed 0 0 0 0 0 0 0 0 --:--:-- 0:00:03 --:--:-- 0 -k disable certificate, but i don't get anything
> 
> Look the files in the folder: `dir` Do you see the file download? Also I am not sure if vcpkg there is an option to download with ignoring the certificate. Also it not a safe option.

So,there is a file,but it looks like broke 
![image](https://user-images.githubusercontent.com/110056050/220962937-4c32ea1e-7bc7-4d90-8fd6-6e45690e27aa.png)

**Comment by VadimS077** [comment]
> > Also i tried the same on my PC and it works. So the problem is local with my laptop
> 
> Is it a laptop from work? Usually the have vpn and configure that it hard to download stuff inside.

no,it's my laptop

**Comment by VadimS077** [comment]
BIG UPDATE: So,why was  this problem i didn't find out. But i reset my PC with new windows and it's working now
